### DISCUSSION TITLE
[Linux] Add Support for native menus

### DISCUSSION THREAD

**Discussion by mzen17** [godotengine/godot-proposals#9942] [GUI]
Currently using Godot 4.3 beta1. When trying to create a native menu on Linux, I get:
[WARNING: Global menus are not supported on this platform], and no menu.

Proposal: A Linux implementation for methods listed at the [docs](https://docs.godotengine.org/en/latest/classes/class_nativemenu.html#class-nativemenu-method-create-menu)

**Comment by bruvzg** [comment]
Unlike on Windows and macOS, there's no such thing as a standard native menu on Linux. Menus are implemented by different widget libraries in different ways.

The closest thing is `com.canonical.dbusmenu` DBus interface (support might be implemented at some point as a part of `StatusIndicator`), but it's limited and probably won't be suitable for general usage.

**Comment by mzen17** [comment]
I was thinking of a widget library implementation (e.g, GTK or QT), since native Linux interface is GTK or QT, and then putting it over Dbus. 

I've seen a many non-GTK applications do this (such as electron apps), so I feel like it is possible, at least, to a certain extent.
